component/identity/src/main/java/org/exoplatform/services/organization/idm/PicketLinkIDMOrganizationServiceImpl.java

```
@@ -229,6 +254,27 @@ public void endRequest(ExoContainer container)
       }
    }
 
+   // Should be used only for non-JTA environment
+   public void recoverFromIDMError(Transaction idmTransaction )
+   {
```

This method has to throw an exception at the end to say that the original operation in top layer has errors and the transaction is rolledback.
We don't have to swallow the exception.